OpenSim/Simulation/Model/PhysicalFrame.cpp

```diff
@@ -88,9 +88,10 @@ void PhysicalFrame::extendConnectToModel(Model& aModel)
 {
     Super::extendConnectToModel(aModel);
 
-    // TODO: Better use name search or more robust method
-    if (upd_geometry(0).getFrameName() == "")
-        upd_geometry(0).setFrameName(getName());
+    // 
+    int idx = getProperty_geometry().findIndexForName("frame_geometry");
+    if (idx != -1 && upd_geometry(idx).getFrameName() == "")
+        upd_geometry(idx).setFrameName(getName());
```

Comment:
@sherm1 As it is today, yes there's a piece of geometry with the name "frame_geometry" that is used to visualize the PhysicalFrame. Whether this will stay or be moved to another class (e.g. Frame) or removed completely seems to be separate from this PR, it's basically how I tested that the API to lookup by name works since we have no other list properties that contain named objects.